src/lib/rest/rest.cpp

```diff
@@ -811,6 +811,18 @@ void firstServicePath(const char* servicePath, char* servicePath0, int servicePa
 
 
 
+/* ****************************************************************************
+*
+* isOriginAllowedForCORS - checks the Origin header of the request and returns
+* true if that Origin is allowed to make a CORS request
+*/
+bool isOriginAllowedForCORS(std::string requestOrigin)
```

Comment:
It should be `const std::string&`, see https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/contribution_guidelines.md#programming-patterns:

> if the function should not be able to modify the object (i.e. "read only"), then const references should be used, e.g. `const BigFish& bf`

Reply:
Thanks for pointing out these mistakes @fgalan, I will fix them quickly.

Reply:
They aren't big issues, but I think is good to fix them so you get familiar with style aspects ;)

Reply:
Yep, we need mistakes to memorize style guidelines :)

Reply:
Fixed in da358b6

Reply:
Just to add some clarification here ...
This is a bit more than just 'style guide' (understanding style guide as a "the way I prefer to see the code, my taste").
This is about avoiding to copy an entire object on the stack for every call to the function, and instead just passing the object by reference.

The style guide says something like:

"An entire object should never ever be sent over the stack."
"If the object is allowed to be altered by the function, use a C pointer"
"If not, use a C++ reference"

The reasoning behind this rule is that the ampersand in the call:
`funcCall(&v):`
**shows** that the variable `v` may be altered, while
`funcCall(v)`
does not.